### PR TITLE
Firefox issue where tree offset is wrong if scrolled horizontally

### PR DESCRIPTION
**Pull request by tomcully**
where tree offset is wrong if scrolled horizontally.

Issue #323 in main repo

### CODE REVIEW

File: source/directives/uiTreeNode.js [hunk 1]
```diff
@@ -261,8 +261,8 @@
                 // check if add it as a child node first
                 // todo decrease is unused
                 decrease = (UiTreeHelper.offset(dragElm).left - UiTreeHelper.offset(placeElm).left) >= config.threshold;
-                targetX = eventObj.pageX - $window.document.body.scrollLeft;
-                targetY = eventObj.pageY - (window.pageYOffset || $window.document.documentElement.scrollTop);
+                targetX = eventObj.pageX - (window.pageXOffset || $window.document.body.scrollLeft || $window.document.documentElement.scrollLeft);
```
Comment (Voles) [inline source/directives/uiTreeNode.js]: Please use the `$window` service.

**Comment by Voles** [comment]
Thanks for this, @tomcully! Can you explain why you need `$window.document.documentElement.scrollLeft` besides `$window.document.body.scrollLeft`?

**Comment by tomcully** [comment]
@Voles np! documentelement.scrollleft is used by some versions of firefox I think. Firefox was getting a '0' for $window.document.body.scrollLeft in some cases. This [Stack Overflow](http://stackoverflow.com/questions/20514596/document-documentelement-scrolltop-return-value-differs-in-chrome) hints that using the combination of the 3 is best cross browser implementation too

**Comment by tomcully** [comment]
@Voles Hi, is everything ok with this? :+1:

**Comment by Voles** [comment]
Sorry for the late reply, @tomcully. I've checked the jQuery source code and they also use `clientTop` and `clientLeft`. I guess we might want to add these too here?

See https://github.com/jquery/jquery/blob/d92310050ca7bf0b33825d64e052f9a8809c3e9e/test/data/jquery-1.9.1.js#L9680

Let me know what you think, I'll do my best follow up as quickly as possible.

**Comment by Voles** [comment]
Thanks for your update! We're close, but I get the error `docElem is not defined` when dragging a node around. Did you mean to put `$window.document.documentElement` in a variable?

**Comment by Voles** [comment]
Thanks for your work, @tomcully!

**Comment by tomcully** [comment]
No probs :+1: Will look at some of the issues on the list when I get a sec :)
resolve #323

**Comment by tomcully** [comment]
@Voles Hey, is there a release planned soon? Just looking to move to the latest version of ui-tree, this change is the only thing we're waiting on for compatibility :+1:

**Pull request merged**